ipapython/dnsutil.py

```diff
@@ -71,7 +71,10 @@ def __str__(self):
 
     def ToASCII(self):
         #method named by RFC 3490 and python standard library
-        return self.to_text().decode('ascii')  # must be unicode string
+        res = self.to_text()
+        if six.PY2:
+            return res.decode('ascii')  # must be unicode string in Python 2
+        return res
```

Comment:
but to_text() returns punycoded values, escaped values as octal numbers, and ascii 32<characters<=127. So I don't think that string can contain something else than ASCII characters < 128